src/modeler/AutoRest.Swagger/Validation/APIVersionPattern.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Logging;
+using AutoRest.Core.Properties;
+using AutoRest.Core.Validation;
+using System.Text.RegularExpressions;
+
+namespace AutoRest.Swagger.Validation
+{
+    /// <summary>
+    /// Validates the version of the swagger. API version must follow the date pattern
+    /// yyyy-MM-dd and allowed prefixes are -preview, -alpha, -beta, -rc, -privatepreview.
+    /// </summary>
+    public class APIVersionPattern : TypedRule<string>
+    {
+        private static readonly Regex VersionRegex = new Regex(@"^(20\d{2})-(0[1-9]|1[0-2])-((0[1-9])|[12][0-9]|3[01])(-(preview|alpha|beta|rc|privatepreview))?$");
```

Comment:
@amarzavery @sarangan12 - This is the required pattern for API version - YYYY-MM-DD[-preview|-alpha|-beta|-rc|-privatepreview]
So, above is not a valid API version